gateway/ln-gateway/src/lightning_manager.rs

```diff
@@ -0,0 +1,67 @@
+use std::fmt::Display;
+use std::sync::Arc;
+
+use bitcoin::Network;
+use fedimint_core::secp256k1::PublicKey;
+use tokio::sync::RwLock;
+
+use super::lightning::{ILnRpcClient, LightningBuilder};
+
+pub struct LightningManager {
+    /// Builder struct that allows the gateway to build a `ILnRpcClient`, which
+    /// represents a connection to a lightning node.
+    pub lightning_builder: Arc<dyn LightningBuilder + Send + Sync>,
+
+    /// The current state of the Gateway.
+    pub state: Arc<RwLock<GatewayState>>,
+}
+
+impl std::fmt::Debug for LightningManager {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("LightningManager")
+            .field("state", &self.state)
+            .finish_non_exhaustive()
+    }
+}
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    Initializing -- begin intercepting HTLCs --> Connected
+///    Initializing -- gateway needs config --> Configuring
+///    Configuring -- configuration set --> Connected
+///    Connected -- load federation clients --> Running
+///    Running -- disconnected from lightning node --> Disconnected
+///    Disconnected -- re-established lightning connection --> Connected
+/// ```
+#[derive(Clone, Debug)]
+pub enum GatewayState {
```

Comment:
Hmm I don't think I agree that this should be managed by `LightningManager`. The `Initializing` and `Configuring` steps have nothing to do with the Lightning node.

Reply:
Should we just remove the `LightningManager` struct then? That just leaves it as a shell around the `LightningBuilder`

Reply:
I think I still like `LightningManager` for the encapsulation of HTLC interception. It might get a little messy with the `FederationManager` though, since that needs to lookup clients. Let's do the `FederationManager` refactor first and see if it makes sense then.